src/app/donation-start/donation-start-form/donation-start-form.component.ts

```diff
@@ -571,6 +579,22 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
   }
 
   async stepChanged(event: StepperSelectionEvent) {
+    if (event.selectedIndex > 0 && !this.donor) {
+      if (event.selectedIndex > 1) {
```

Comment:
I was trying to understand why selected index 1 is treated as a special case, but I didn't quite get there. Could we add a comment?

Also if 1 does need to be a special case it might make the code clear to deal with the case of `event.selectedIndex === 1` first.